include/chemfiles/Atom.hpp

```diff
@@ -178,6 +178,14 @@ class CHFL_EXPORT Atom final {
     /// @example{tests/doc/atom/property.cpp}
     optional<const Property&> get(const std::string& name) const;
 
+    /// Get the constant beginning iterator for the atom's properties.
+    optional<std::unordered_map<std::string, Property>::const_iterator> properties_begin() const {
```

Comment:
We can use a `typedef/using` to define a `properties_iterator` for clarity instead of using `std::unordered_map<std::string, Property>::const_iterator`

Reply:
Also, all the function will need documentation and examples